packages/components/src/templates/next/components/complex/Video/Video.tsx

```diff
@@ -0,0 +1,105 @@
+import type { VideoProps } from "~/interfaces"
+import { VIDEO_EMBED_URL_PATTERN } from "~/utils/validation"
+import { ComponentContent } from "../../internal/customCssClass"
+
+const VALID_VIDEO_DOMAINS = {
+  youtube: [
+    "www.youtube.com",
+    "www.youtube-nocookie.com",
+    "youtube.com",
+    "youtube-nocookie.com",
+  ],
+  vimeo: ["player.vimeo.com"],
+  fbvideo: ["www.facebook.com"],
+}
+
+const isValidVideoUrl = (url: string) => {
+  if (!url) {
+    return false
+  }
+
+  try {
+    const urlObject = new URL(url)
+    const allValidVideoDomains = Object.values(VALID_VIDEO_DOMAINS).flat()
+
+    return (
+      allValidVideoDomains.includes(urlObject.hostname) &&
+      new RegExp(VIDEO_EMBED_URL_PATTERN).test(url)
+    )
+  } catch (_) {
+    return false
+  }
+}
+
+// NOTE: We are only using the privacy-enhanced mode of YouTube embeds
+const getPrivacyEnhancedYouTubeEmbedUrl = (urlObject: URL) => {
+  if (urlObject.hostname === "www.youtube.com") {
+    urlObject.hostname = "www.youtube-nocookie.com"
+  }
+
+  const { pathname, searchParams } = urlObject
+
+  if (pathname.startsWith("/embed/")) {
+    return urlObject.toString()
+  } else if (pathname.startsWith("/watch")) {
+    const videoId = searchParams.get("v")
+    if (!videoId) {
+      return ""
+    }
+
+    urlObject.pathname = `/embed/${videoId}`
+    urlObject.search = ""
+
+    return urlObject.toString()
+  }
+}
+
+// NOTE: We are setting a do-not-track attribute on Vimeo embeds
+// Ref: https://developer.vimeo.com/api/oembed/videos
+const getPrivacyEnhancedVimeoEmbedUrl = (urlObject: URL) => {
+  urlObject.searchParams.set("dnt", "true")
+  return urlObject.toString()
+}
+
+const getSanitizedEmbedUrl = (url: string) => {
+  if (!isValidVideoUrl(url)) {
+    return ""
+  }
+
+  const urlObject = new URL(url)
+
+  if (VALID_VIDEO_DOMAINS.youtube.includes(urlObject.hostname)) {
+    return getPrivacyEnhancedYouTubeEmbedUrl(urlObject)
+  } else if (VALID_VIDEO_DOMAINS.vimeo.includes(urlObject.hostname)) {
+    return getPrivacyEnhancedVimeoEmbedUrl(urlObject)
+  } else if (VALID_VIDEO_DOMAINS.fbvideo.includes(urlObject.hostname)) {
+    return urlObject.toString()
+  }
+
+  return ""
+}
+
+export const Video = ({ title, url }: VideoProps) => {
+  if (!url || !isValidVideoUrl(url)) {
```

Comment:
nit - no need `!url` since it's being handled by `isValidVideoUrl`

Reply:
Updated in eb20eb67e0afb85ceacd76bee3a32dba5d2ced92